### DISCUSSION TITLE
Transmission 4.0.3 Linux wont connect to any peers

### DISCUSSION THREAD

**Discussion by j74s** [transmission/transmission#8436] [Support]
### What is the issue?

So I got the update from transmission 3 to 4.0.3 on Fedora 37, and as the title says, it wont connect to any peers.
Version 3 works perfectly or any other bt client for that mater, I tried changing settings and what not to no avail.
From the debug log:

`|qui mar  9 21:01:50 2023|Information|port-forwarding-natpmp.cc:177|Port 51413 forwarded successfully|
|qui mar  9 21:01:50 2023|Debug|port-forwarding-natpmp.cc:38|readnatpmpresponseorretry succeeded (0)|
|qui mar  9 21:01:45 2023|Debug|Fedora-Workstation-Live-x86_64-37 at torrent.fedoraproject.org:6969|tracker knows of 379 seeders and 7 leechers and gave a list of 80 peers.`

After this it just sits there and does not connect to peers. Again, version 3 still works perfectly, I also tested version 4.0.3 on a windows 10 install an it also works, it seems to be related to version 4 and Linux.

### Which application of Transmission?

GTK+ app on Linux, BSD, etc.

### Which version of Transmission?

4.0.3

**Comment by shivanshu3** [comment]
I saw this too, and I think it was fixed in #5460 
I haven't tried running a nightly build though, so I'm not 100% sure. I hope 4.0.4 gets released soon with this patch, it's been a while now 😟

**Comment by j74s** [comment]
> I saw this too, and I think it was fixed in #5460 I haven't tried running a nightly build though, so I'm not 100% sure. I hope 4.0.4 gets released soon with this patch, it's been a while now worried

Thank you very much for letting me know. I will test it as soon as its released.

**Comment by tearfur** [comment]
@j74s How is it in `4.0.4`?

**Comment by j74s** [comment]
@tearfur There are no official packages for fedora 38 yet, so installed an rpm from the rawhide repo, and unfortunately it still wont connect to any pears. This is very disappointing.

`qua ago 30 19:16:43 2023	Debug	Fedora-Workstation-Live-ppc64le-38	setting [136.35.138.88]:53269 do_purge flag because we got an ERANGE, EMSGSIZE, or ENOTCONN error
qua ago 30 19:16:42 2023	Debug	Fedora-Workstation-Live-ppc64le-38	setting [136.35.138.88]:53269 do_purge flag because we got an ERANGE, EMSGSIZE, or ENOTCONN error
qua ago 30 19:14:38 2023	Debug	Fedora-Workstation-Live-ppc64le-38 at torrent.fedoraproject.org:6969	tracker knows of 55 seeders and 1 leechers and gave a list of 56 peers.
qua ago 30 19:14:36 2023	Debug	Fedora-Workstation-Live-ppc64le-38	Verification is done. It took 0 seconds to verify 2069308456 bytes (2069308456 bytes per second)
qua ago 30 19:14:36 2023	Debug	Fedora-Workstation-Live-ppc64le-38	verifying torrent...
qua ago 30 19:14:36 2023	Debug	Fedora-Workstation-Live-ppc64le-38	Verification is done. It took 0 seconds to verify 2069308456 bytes (2069308456 bytes per second)
qua ago 30 19:14:36 2023	Debug	Fedora-Workstation-Live-ppc64le-38	verifying torrent...`

`qua ago 30 19:21:53 2023	Debug	Fedora-Workstation-Live-ppc64le-38	setting [136.35.138.88]:57922 do_purge flag because we got an ERANGE, EMSGSIZE, or ENOTCONN error
qua ago 30 19:21:53 2023	Debug	Fedora-Workstation-Live-ppc64le-38	setting [136.35.138.88]:57922 do_purge flag because we got an ERANGE, EMSGSIZE, or ENOTCONN error
qua ago 30 19:21:31 2023	Debug	Fedora-Workstation-Live-ppc64le-38	setting [83.31.13.34]:40885 do_purge flag because we got an ERANGE, EMSGSIZE, or ENOTCONN error
qua ago 30 19:21:31 2023	Debug	Fedora-Workstation-Live-ppc64le-38	setting [83.31.13.34]:40885 do_purge flag because we got an ERANGE, EMSGSIZE, or ENOTCONN error
qua ago 30 19:20:56 2023	Debug	Fedora-Workstation-Live-ppc64le-38	setting [134.101.166.253]:63734 do_purge flag because we got an ERANGE, EMSGSIZE, or ENOTCONN error
qua ago 30 19:20:56 2023	Debug	Fedora-Workstation-Live-ppc64le-38	setting [134.101.166.253]:63734 do_purge flag because we got an ERANGE, EMSGSIZE, or ENOTCONN error
qua ago 30 19:20:09 2023	Debug	Fedora-Workstation-Live-ppc64le-38	setting [83.31.13.34]:34617 do_purge flag because we got an ERANGE, EMSGSIZE, or ENOTCONN error
qua ago 30 19:20:09 2023	Debug	Fedora-Workstation-Live-ppc64le-38	setting [83.31.13.34]:34617 do_purge flag because we got an ERANGE, EMSGSIZE, or ENOTCONN error
qua ago 30 19:19:30 2023	Debug	Fedora-Workstation-Live-ppc64le-38	setting [136.35.138.88]:30017 do_purge flag because we got an ERANGE, EMSGSIZE, or ENOTCONN error
qua ago 30 19:19:29 2023	Debug	Fedora-Workstation-Live-ppc64le-38	setting [136.35.138.88]:30017 do_purge flag because we got an ERANGE, EMSGSIZE, or ENOTCONN error
qua ago 30 19:18:32 2023	Debug	Fedora-Workstation-Live-ppc64le-38	setting [134.101.166.253]:42140 do_purge flag because we got an ERANGE, EMSGSIZE, or ENOTCONN error
qua ago 30 19:18:32 2023	Debug	Fedora-Workstation-Live-ppc64le-38	setting [134.101.166.253]:42140 do_purge flag because we got an ERANGE, EMSGSIZE, or ENOTCONN error
qua ago 30 19:18:06 2023	Debug	Fedora-Workstation-Live-ppc64le-38	setting [136.35.138.88]:2318 do_purge flag because we got an ERANGE, EMSGSIZE, or ENOTCONN error
qua ago 30 19:18:05 2023	Debug	Fedora-Workstation-Live-ppc64le-38	setting [136.35.138.88]:2318 do_purge flag because we got an ERANGE, EMSGSIZE, or ENOTCONN error
qua ago 30 19:16:43 2023	Debug	Fedora-Workstation-Live-ppc64le-38	setting [136.35.138.88]:53269 do_purge flag because we got an ERANGE, EMSGSIZE, or ENOTCONN error
qua ago 30 19:16:42 2023	Debug	Fedora-Workstation-Live-ppc64le-38	setting [136.35.138.88]:53269 do_purge flag because we got an ERANGE, EMSGSIZE, or ENOTCONN error`

**Comment by tearfur** [comment]
Well I test my Transmission code on a Fedora 38 machine regularly and did not have any of your problems, so it is unlikely that there's something wrong with Transmission 4. But then everything else works for you so this is even more weird...

You logs aren't showing anything out of the ordinary, so I can only make some wild guesses...

What happens if you reset Transmission's settings? You can do so by moving the config directory to another location using `mv`. By default, the config directory is at `~/.config/transmission-daemon`.

**Comment by j74s** [comment]
@tearfur I will try that and report back. 
What bugs me is that Transmission 3 works fine on Fedora 38 and Transmission 4 also works fine on a Windows 10 virtual machine  on a Fedora 38 host. Is there any way I can debug this further? Thank you.

**Comment by j74s** [comment]
@tearfur  Deleted config and tested, still not working. :(

**Comment by Pentaphon** [comment]
@j74s how is 4.0.6?

**Comment by j74s** [comment]
@Pentaphon Unfortunately, still not working on Fedora 40, cant connect to trackers nor peers. But get this, on a windows 10 vm inside the same Fedora 40 host, it works. Same network same Transmission version. I'm baffled. Windows version works nicely bu on Linux it will not. Still 3.0, works well on Linux.

**Comment by Pentaphon** [comment]
> Unfortunately, still not working on Fedora 40

Sure there's no firewall software in your way? Tried another distro on another machine? I use 4.0.6 on Pop!OS and Windows 11 and it works fine.

**Comment by j74s** [comment]
Well, if there's a firewall it would have to be upstream  and then Transmission 3.0 would no work either. But I will test other Linux distros, again, on windows, Transmission 4.x, works fine. Ill try to narrow it down.
